scripts/create_broker_secret.py

```diff
@@ -104,17 +124,49 @@ def format_config(config):
         del config['metadata'][key]
     return yaml.dump(config)
 
-def fqname(apb):
-    registries = {'docker.io': 'dh'}
-    registry, org, end = apb.split('/')
 
-    if ":" in end:
-        image, tag = end.split(":")
-    else:
-        image = end
-        tag = 'latest'
+def broker_url():
+    return 'https://' + runcmd('oc get routes --no-headers').split()[1]
 
-    return '-'.join([registries[registry], org, image, tag])
+
+def get_all_apbs():
+    response = requests.get(broker_url() + '/v2/catalog', verify=False)
+    return response.json()['services']
+
+
+
+def fqname(apb):
+    parts = apb.split('/')
+    if len(parts) == 3:
+        # Chop the service, we don't know what it will be translated to in the broker
+        parts = parts[1:]
+    if ':' in parts[-1]:
```

Comment:
Would this if statement ever be skipped? If it can, shouldn't latest be added to the end of parts?